internal/envoy/v3/route.go

```diff
@@ -194,7 +194,9 @@ func PathRouteMatch(pathMatchCondition dag.MatchCondition) *envoy_route_v3.Route
 		case dag.PrefixMatchSegment:
 			return &envoy_route_v3.RouteMatch{
 				PathSpecifier: &envoy_route_v3.RouteMatch_PathSeparatedPrefix{
-					PathSeparatedPrefix: c.Prefix,
+					// Trim trailing slash as PathSeparatedPrefix expects
+					// no trailing slashes.
+					PathSeparatedPrefix: strings.TrimSuffix(c.Prefix, "/"),
```

Comment:
I think it makes sense to use `strings.TrimRight` here to handle removing any number of trailing slashes, just to be safe since there is no particular validation on the field